Game/Source/GDKTestGyms/Private/LatencyTracer.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LatencyTracer.h"
+
+#include "Engine/World.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogLatencyTracer);
+
+namespace LatencyTracerInternal
+{
+	SpatialGDK::SpatialEventTracer* GetEventTracerFromNetDriver(const UWorld* World)
+	{
+		if (USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(World->GetNetDriver()))
+		{
+			return NetDriver->Connection->GetEventTracer();
+		}
+
+		return nullptr;
+	}
+}
+
+ULatencyTracer* ULatencyTracer::GetOrCreateLatencyTracer(UObject* WorldContextObject)
+{
+	UWorld* World = GEngine->GetWorldFromContextObject(WorldContextObject, EGetWorldErrorMode::ReturnNull);
+	if (World == nullptr)
+	{
+		World = GWorld;
+	}
+
+	if (UGDKTestGymsGameInstance* GameInstance = World->GetGameInstance<UGDKTestGymsGameInstance>())
+	{
+		if (GameInstance->GetTracer() != nullptr)
+		{
+			return GameInstance->GetTracer();
+		}
+	}
+	UE_LOG(LogLatencyTracer, Error, TEXT("Unable to get the game instance tracer object, creating a new one but results may not be as expected."));
```

Comment:
`GetOrCreateLatencyTracer` The name seems to imply that creating latency tracer is not an error?

Reply:
Indeed, reduced to a log.